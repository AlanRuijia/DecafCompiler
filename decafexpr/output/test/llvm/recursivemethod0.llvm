; ModuleID = 'Test'

declare void @print_int(i32)

define void @test(i32) {
entry:
  %y = alloca i32
  store i32 undef, i32* %y
  %x = load i32, i32* %x
  %y1 = load i32, i32* %y
  %AddTmp = add i32 %x, %y1
  call void @test(i32 %AddTmp)
  store i32 %0, i32* %x
  ret void
}

define i32 @main() {
entry:
  %x = alloca i32
  store i32 1, i32* %x
  %x1 = load i32, i32* %x
  call void @test(i32 %x1)
  ret i32 0
}
