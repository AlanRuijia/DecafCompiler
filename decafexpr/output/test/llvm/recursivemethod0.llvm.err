Assertion failed: (getType()->isIntOrIntVectorTy() && "Tried to create an integer operation on a non-integer type!"), function init, file /private/tmp/llvm3820160316-8183-11u7j5k/llvm-3.8.0.src/lib/IR/Instructions.cpp, line 1944.
