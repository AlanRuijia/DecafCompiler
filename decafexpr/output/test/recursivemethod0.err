output prefix: /Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod0
llvmas: /usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llvm-as
llc: /usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llc
cc: gcc
codegen: /Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/answer/decafexpr
stdlib: /Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/answer/decaf-stdlib.c
generating llvm code:/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/answer/decafexpr ... ok
; ModuleID = 'Test'

declare void @print_int(i32)

define void @test(i32) {
entry:
  %y = alloca i32
  store i32 undef, i32* %y
  %x = load i32, i32* %x
  %y1 = load i32, i32* %y
  %AddTmp = add i32 %x, %y1
  call void @test(i32 %AddTmp)
  store i32 %0, i32* %x
  ret void
}

define i32 @main() {
entry:
  %x = alloca i32
  store i32 1, i32* %x
  %x1 = load i32, i32* %x
  call void @test(i32 %x1)
  ret i32 0
}
assembling to bitcode:/usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llvm-as "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod0.llvm" -o "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod0.llvm.bc" ... failed (1)
/usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llvm-as: /Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod0.llvm:9:3: error: instruction forward referenced with type 'i32*'
  %x = load i32, i32* %x
  ^
converting to native code:/usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llc "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod0.llvm.bc" -o "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod0.llvm.s" ... ok
linking:gcc -o "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod0.llvm.exec" "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod0.llvm.s" "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/answer/decaf-stdlib.c" ... failed (1)
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
running:/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod0.llvm.exec ...Traceback (most recent call last):
  File "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/answer/llvm-run", line 172, in <module>
    result &= run("running", "%s.llvm.exec" % (out_prefix), ".run", None, out_prefix)
  File "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/answer/llvm-run", line 90, in run
    retval = subprocess.call(shlex.split(cmd), stdin=infile, stdout=outoutfile, stderr=outerrfile)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 522, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
