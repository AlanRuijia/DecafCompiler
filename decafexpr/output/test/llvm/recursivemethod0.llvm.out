Debug message: Generating extern function...
Arg Type: 17 
Inserting method decl into symbol table...
Inserting block into symbol table...
Inserting method decl into symbol table...
Inserting block into symbol table...
Debug message: Generating program...
Debug message: Generating package...
Debug message: Generating FieldDecls...
Debug message: Generating MethodDecl...
Method Block CodeGen...
Block CodeGen......
Codegen for Var definition(Typed Symbol)...
Finish Codegen for Var definition(Typed Symbol)...
Defind variable in line 5 : y
Starting NumberExpr Codegen
Finished NumberExpr Codegen
Starting NumberExpr Codegen
Finished NumberExpr Codegen
OpId 3
Debug message: Generating method call... Calling test
ArgumentListSize = 1
Argument Type: Codegen for RValue x
Codegen for RValue y
OpId 0
